### PR TITLE
fix: improve connection closing

### PR DESCRIPTION
**Pull request by gssbzn**
## Description

Bringing this from the ops manager client, in some cases we some extra bytes may be left in the body, this way we make sure to read it all so the connection can be reused 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by themantissa — APPROVED** [review]
LGTM - will add DoU for visibility. Thank you!

**Comment by themantissa** [comment]
@coderGo93 and @leofigy added as reviewers for FYI, just need one of you to approve.

**Comment by gssbzn** [comment]
@coderGo93 @leofigy just a ping on this as we may do a release this week and I'd like to include this one

File: mongodbatlas/mongodbatlas.go [hunk 1]
```diff
@@ -387,9 +387,18 @@ func (c *Client) Do(ctx context.Context, req *http.Request, v interface{}) (*Res
...
-			err = rerr
+		// Ensure the response body is fully read and closed
+		// before we reconnect, so that we reuse the same TCP connection.
+		// Close the previous response's body. But read at least some of
+		// the body so if it's small the underlying TCP connection will be
+		// re-used. No need to check for errors: if it fails, the Transport
+		// won't reuse it anyway.
+		const maxBodySlurpSize = 2 << 10
+		if resp.ContentLength == -1 || resp.ContentLength <= maxBodySlurpSize {
+			_, _ = io.CopyN(ioutil.Discard, resp.Body, maxBodySlurpSize)
```
Comment (leofigy) [inline mongodbatlas/mongodbatlas.go:398]: Nice !

**Review by leofigy — APPROVED** [review]

**Pull request merged**